plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/config.py

```diff
@@ -17,12 +18,86 @@ class Direction(Enum):
     maximize = 2
 
 
-class Sampler(Enum):
-    tpe = 1
-    random = 2
-    cmaes = 3
-    nsgaii = 4
-    motpe = 5
+@dataclass
+class TPESamplerConfig:
```

Comment:
You can create a superclass for all of those to get better type safety.
```python
@dataclass
class SamplerConfig:
  _target_: str = MISSING

@dataclass
class TPESamplerConfig(SamplerConfig):
  ...

@dataclass
class OptunaSweeperConf:
    _target_: str = "hydra_plugins.hydra_optuna_sweeper.optuna_sweeper.OptunaSweeper"
    ...
    sampler: SamplerConfig = MISSING
```

Reply:
will do